src/Grpc.StatusProto/README.md

```diff
@@ -0,0 +1,256 @@
+                    {
+                        Console.WriteLine($"\tKey: {md.Key}, Value: {md.Value}");
+                    }
+                    break;
+
+                case BadRequest badRequest:
+                    Console.WriteLine("BadRequest:");
+                    foreach (BadRequest.Types.FieldViolation fv in badRequest.FieldViolations)
+                    {
+                        Console.WriteLine($"\tField: {fv.Field}, Description: {fv.Description}");
+                    }
+                    break;
+
+                // Other cases handled here ...
+            }
+        }
+    }
+
+```
+
+## Returning errors within gRPC streams
+
+The model described above allows you to return an error status when the gRPC call finishes. Sometimes
+when using gRPC streams that allow clients and servers to send multiple messages in a single call
+you may wish to return a status without terminating the call.
+
+To do this the definition of the stream of messages returned from the service should itself contain
+a `google.rpc.Status` message, for example:
+
+```protobuf
+service WidgetLookupProvider {
+    rpc streamingLookup(stream WidgetReq) returns (stream WidgetRsp) {}
+}
+
+message WidgetReq {
+    string widget_name = 1;
+}
+
+message WidgetRsp {
+    oneof message{
+        // details when ok
+        string widget_details = 1;
+        // or error details
+        google.rpc.Status status = 2;
+   }   
+}
+```
+
+Note: the  [status.proto](https://github.com/googleapis/googleapis/blob/master/google/rpc/status.proto)
+and [error_details.proto](https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto)
+files are not currently provided in any NuGet packages. The text for those proto files can be copied from
+the given links and included in your project.
+
+Example server code fragment:
+```C#
+ while (await requestStream.MoveNext())
+{
+    var request = requestStream.Current;
+    var response = new WidgetRsp();
+
+    // ... process the request ...
+
+    // to return an error
+    if (error)
+    {
+        response.Status = new Google.Rpc.Status { /* ... */ };
+    } else
+    {
+        response.WidgetDetails = "the details";
+    }
+}
+```
+
+Example client code fragment:
+```C#
+
+// reading the responses
+var responseReaderTask = Task.Run(async () =>
+{
+    while (await call.ResponseStream.MoveNext())
+    {
+        var rsp = call.ResponseStream.Current;
+        switch (rsp.MessageCase)
+        {
+            case WidgetRsp.MessageOneofCase.WidgetDetails:
+                // ... processes the details ...
+                break;
+            case WidgetRsp.MessageOneofCase.Status:
+                // ... handle the error ...
+                break;
+        }
+    }
+});
+
+// sending the requests
+foreach (var request in requests)
+{
+    await call.RequestStream.WriteAsync(request);
+}
+```
+
+
+## See also
+* [Richer error model](https://grpc.io/docs/guides/error/#richer-error-model)
```

Comment:
perhaps you can link directly to https://cloud.google.com/apis/design/errors#error_model somewhere in the README since that documentation has quite a lot of useful info.